extensions/search-manifest/src/uploadToAtlas/deleteStale.ts

```
@@ -20,10 +22,21 @@ export const deleteStaleDocuments = async ({
   };
 };
 
-export const deleteStaleProperties = async (searchProperty: string) => {
-  const documentsColl = await getDocumentsCollection();
-  console.debug(`Removing all documents with stale property ${searchProperty}`);
+export const deleteStaleProperties = async (
+  searchProperty: string,
+  connectionInfo: SearchClusterConnectionInfo,
+) => {
+  const documentsColl = await getDocumentsCollection(connectionInfo);
+  console.info(`Removing all documents with stale property ${searchProperty}`);
   const query = { searchProperty: { $regex: searchProperty } };
-  const status = await documentsColl?.deleteMany(query);
-  return status;
+  try {
+    const status = await documentsColl?.deleteMany(query);
```

Instead of using optional chaining, I think we should throw an error in the `getDocumentsCollection` function if we don't have a `documentsColl` object. If we don't have that, then this code doesn't work, and I think it makes sense to have that error occur where it happens.

---

I just removed the optional chaining so an error will be thrown if we try to run `deleteMany()` on a nonexistent collection